### PR TITLE
add support for [\n] character format parameter inside config file

### LINKED ISSUE #61: Changing fromat from configuration file does not recognize '\n' character
As an example: if you put`format=%s\n%b` in the config file, the notification result wold be   `Summary\nBody` (those are two value that i'm using for testing).
Debugging the line that mako gets from the config file, you get that the `\` char is escaped.
gdb output (breakpoint at config.c:378):
```
(gdb) p eq + 1
$1 = 0x555555572387 "%s\\n%b"
```

### PR DESCRIPTION
**Pull request by masozzi**
This fixes #61 
As a consequence also the char `\` needs to be escaped.

i'll update the `mako.1.sdc` file once the modification are approved

### CODE REVIEW

File: types.c [hunk 1]
```diff
@@ -123,3 +124,55 @@ bool parse_directional(const char *string, struct mako_directional *out) {
...
+	memset(token, 0, token_max_length);
+	size_t token_location = 0;
+
+	enum mako_parse_state state = MAKO_PARSE_STATE_NORMAL;
+	const char *location = string;
+
+	char ch;
+	while ((ch = *location++) != '\0') {
+		switch (state) {
+			case MAKO_PARSE_STATE_ESCAPE:
```
Comment (ascent12) [inline types.c]: I think the style is to not indent case labels.

File: types.c [hunk 2]
```diff
@@ -123,3 +124,55 @@ bool parse_directional(const char *string, struct mako_directional *out) {
 	free(components);
 	return true;
 }
+
+bool parse_format(const char *string, char **out) {
+	int token_max_length = strlen(string) + 1;
```
Comment (ascent12) [inline types.c]: size_t is more correct here.

File: types.c [hunk 3]
```diff
@@ -123,3 +124,55 @@ bool parse_directional(const char *string, struct mako_directional *out) {
...
+	char token[token_max_length];
+	memset(token, 0, token_max_length);
+	size_t token_location = 0;
+
+	enum mako_parse_state state = MAKO_PARSE_STATE_NORMAL;
+	const char *location = string;
+
+	char ch;
+	while ((ch = *location++) != '\0') {
```
Comment (ascent12) [inline types.c]: I think a normal for loop would be better.

File: types.c [hunk 4]
```diff
@@ -123,3 +124,55 @@ bool parse_directional(const char *string, struct mako_directional *out) {
...
+		switch (state) {
+			case MAKO_PARSE_STATE_ESCAPE:
+				switch (ch) {
+					case 'n':
+						token[token_location] = '\n';
+						++token_location;
+						break;
+
+					case '\\':
+					default:
```
Comment (ascent12) [inline types.c]: I'm not sure if you want to ignore unknown escape sequences or not. Sometimes that can mess with the config file's forwards compatibility.
That's up to @emersion though.

**Comment by emersion** [comment]
Thanks for your PR!

I agree with @ascent12's comments (thanks!).

>I'm not sure if you want to ignore unknown escape sequences or not. Sometimes that can mess with the config file's forwards compatibility.

I'm not super-concerned about this. Maybe keeping unknown escaped sequences as-is would be a little bit better.

**Comment by emersion** [comment]
Thanks!

**Comment by masozzi** [comment]
sorry, i should have pinged you.
by the way, thanks for the help!

**Pull request merged**